system/CompletionEvent_tests.cpp

```diff
@@ -276,6 +279,42 @@ void try_synchronizing_spawns() {
   BOOST_CHECK_EQUAL(total, N);
 }
 
+void try_iterative_spmd_gce() {
+
+  // set up re-enrolling across iterations
+  gce.enroll(cores());
+  gce.reenroll_on_complete();
```

Comment:
I don't love this interface. When I first saw it, I assumed the order of the two didn't matter (like, it always re-enrolls any completes or something), but that assumption turned out to be wrong. Can we just create a version of `enroll` that auto-reenrolls? Like: `gce.enroll_recurring(cores())` which would roll up both of those methods into one?